### DISCUSSION TITLE
Nordic Semiconductor nRF52840 chips with ZBOSS Zigbee stack now usable as Zigbee Coordinator for the ZHA integration via new radio library for zigpy

### DISCUSSION THREAD

**Discussion by Hedda** [mercenaruss/zigstar_gateways#29] [General]
FYI, you can now use Nordic Semi nRF52840 serial modules as Zigbee Coordinator for Homa Assistant's [ZHA integration](https://www.home-assistant.io/integrations/zha) via zigpy-zboss:

https://github.com/kardia-as/zigpy-zboss

zigpy-zboss is still unofficial and bleeding-edge experimental so need to manually add (i.e. hack) radio library into the `zha` component:

https://github.com/home-assistant/core/tree/dev/homeassistant/components/zha

Flashed with ZBOSS NCP Host then radio type `zboss` should work like any Zigbee Coordinator adapter + add device serial path to ZHA:
 
https://www.home-assistant.io/integrations/zha

For development discussion see:

https://github.com/zigpy/zigpy/issues/394

and for end-user discussion see his Home Assistant community forum thread:

https://community.home-assistant.io/t/nordic-semi-nrf52840-chips-with-zboss-zigbee-stack-now-usable-as-zigbee-coordinator-for-the-zha-integration-via-new-radio-library-for-zigpy/570196

This new radio library for zigpy should make usable to use either nRF52840 USB dongle (or nRF52840 dev kits) with the correct firmware.

https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dongle

(or https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dk)

nRF52840 Dongle only costs around $10 USD today which makes it a very affordable Zigbee Coordinator for the ZHA integration.

ZigStar hardware developer could now technically make network-attached Zigbee Coordinator Ethernet adapters using ESP32 module:

https://github.com/mercenaruss/zigstar_gateways

Anyway, ZBOSS Zigbee stack support has the potential to be a big deal since its firmware is open-source and is also hardware independent:

https://dsr-zoi.com

https://dsr-iot.com

PS: Off-topic but interesting for future prospect is that is and when this zigpy-zboss library gets developed further then it could able for developers to port it for use with ESP32-C6 (and ESP32-H2) module with ESPHome as native ESP32 Zigbee Coordinator adapter for ZHA: 

https://github.com/esphome/feature-requests/issues/1397

https://github.com/zigpy/zigpy/issues/1052

**Comment by mercenaruss** [comment]
@Hedda thank you for this information. Will take a look in following weeks.

**Comment by mercenaruss** [comment]
@Hedda is any chance that you can contact me?I want to discuss some things. https://zig-star.com/about/
Use Discord or Telegram if possible.
